providers/gitlab.go

```diff
@@ -183,6 +185,49 @@ func (p *GitLabProvider) verifyGroupMembership(userInfo *gitlabUserInfo) error {
 	return fmt.Errorf("user is not a member of '%s'", p.Group)
 }
 
+type gitlabProjectInfo struct {
+	Name     string `json:"name"`
+	Archived bool   `json:"archived"`
+}
+
+func (p *GitLabProvider) getProjectInfo(ctx context.Context, s *sessions.SessionState) (*gitlabProjectInfo, error) {
+	endpointURL := &url.URL{
+		Scheme: p.LoginURL.Scheme,
+		Host:   p.LoginURL.Host,
+		Path:   "/api/v4/projects/",
+	}
+	endpoint := endpointURL.String() + url.QueryEscape(p.Project)
+	fmt.Println("project: ", endpoint)
+	req, err := http.NewRequestWithContext(ctx, "GET", endpoint, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create project info request: %v", err)
+	}
+	req.Header.Set("Authorization", "Bearer "+s.AccessToken)
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to perform project info request: %v", err)
+	}
+	var body []byte
+	body, err = ioutil.ReadAll(resp.Body)
+	resp.Body.Close()
```

Comment:
I'm on a warpath for unhandled closes today to eventually get to a clean Gosec 😄  -- can we handle the error response from this?